File: java/src/test/java/com/kcl/LoadSettingsFilesTest.java
package com.kcl;

import org.junit.Assert;
import org.junit.Test;

import com.kcl.api.API;
import com.kcl.api.Spec.LoadSettingsFilesArgs;
import com.kcl.api.Spec.LoadSettingsFilesResult;

public class LoadSettingsFilesTest {
    @Test
    public void testLoadSettingsFile() throws Exception {
        API api = new API();
        LoadSettingsFilesArgs args = LoadSettingsFilesArgs.newBuilder().addFiles("./src/test_data/settings/kcl.yaml")
                .build();
        LoadSettingsFilesResult result = api.loadSettingsFiles(args);
        Assert.assertEquals(result.getKclCliConfigs().getFilesCount(), 0);
        Assert.assertEquals(result.getKclCliConfigs().getStrictRangeCheck(), true);
        Assert.assertEquals(result.getKclOptions(0).getKey(), "key");
        Assert.assertEquals(result.getKclOptions(0).getValue(), "\"value\"");
    }
}

File: kotlin/src/test/kotlin/com/kcl/LoadSettingsFilesTest.kt
package com.kcl

import com.kcl.api.API
import com.kcl.api.loadSettingsFilesArgs
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.Assertions.assertEquals

class LoadSettingsFileTest {
    companion object {
        private const val TEST_FILE = "./src/test_data/settings/kcl.yaml"
    }

    @Test
    fun testLoadSettingsFilesApi() {
        val args = loadSettingsFilesArgs { files += TEST_FILE }
        val api = API()
        val result = api.loadSettingsFiles(args)
        assertEquals(result.kclCliConfigs.filesCount, 0)
        assertEquals(result.kclCliConfigs.strictRangeCheck, true)
        assertEquals(result.kclOptionsList[0].key, "key")
        assertEquals(result.kclOptionsList[0].value, "\"value\"")
    }
}
